airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumer.kt

```diff
@@ -51,8 +55,30 @@ class AsyncStreamConsumer(
     private val defaultNamespace: Optional<String>,
     private val flushFailure: FlushFailure = FlushFailure(),
     workerPool: ExecutorService = Executors.newFixedThreadPool(5),
-    private val airbyteMessageDeserializer: AirbyteMessageDeserializer,
+    private val airbyteMessageDeserializer: AirbyteMessageDeserializer =
+        AirbyteMessageDeserializer(),
 ) : SerializedAirbyteMessageConsumer {
+    constructor(
+        outputRecordCollector: Consumer<AirbyteMessage>,
+        onStart: OnStartFunction,
+        onClose: OnCloseFunction,
+        onFlush: DestinationFlushFunction,
+        catalog: ConfiguredAirbyteCatalog,
+        bufferManager: BufferManager,
+        defaultNamespace: Optional<String>,
+        workerPool: ExecutorService
+    ) : this(
+        outputRecordCollector = outputRecordCollector,
+        onStart = onStart,
+        onClose = onClose,
+        onFlush = onFlush,
+        catalog = catalog,
+        bufferManager = bufferManager,
+        defaultNamespace = defaultNamespace,
+        flushFailure = FlushFailure(),
+        workerPool = workerPool,
+    )
```

Comment:
I aggressively removed the overloaded constructors hoping the 2 usages were already in Kotlin with defaults injected. Here it is yet another Java usage 🤦‍♂️

Reply:
yep, I noticed your removal 😠 😏